generators/php/sdk/src/endpoint/request/WrappedEndpointRequestGenerator.ts

```diff
@@ -0,0 +1,113 @@
+import { SdkGeneratorContext } from "../../SdkGeneratorContext";
+
+export declare namespace WrappedEndpointRequestGenerator {
+    export interface Args {
+        serviceId: ServiceId;
+        wrapper: SdkRequestWrapper;
+        context: SdkGeneratorContext;
+        endpoint: HttpEndpoint;
+    }
+}
+
+export class WrappedEndpointRequestGenerator extends FileGenerator<
+    PhpFile,
+    SdkCustomConfigSchema,
+    SdkGeneratorContext
+> {
+    private wrapper: SdkRequestWrapper;
+    private serviceId: ServiceId;
+    private endpoint: HttpEndpoint;
+    private classReference: php.ClassReference;
+    private location: FileLocation;
+
+    public constructor({ wrapper, context, serviceId, endpoint }: WrappedEndpointRequestGenerator.Args) {
+        super(context);
+        this.wrapper = wrapper;
+        this.serviceId = serviceId;
+        this.endpoint = endpoint;
+        this.classReference = this.context.getRequestWrapperReference(this.serviceId, this.wrapper.wrapperName);
+        this.location = this.context.getLocationForWrappedRequest(this.serviceId);
+    }
+
+    protected doGenerate(): PhpFile {
+        const class_ = php.class_({
+            ...this.classReference
+        });
+
+        for (const query of this.endpoint.queryParameters) {
+            const type = query.allowMultiple
+                ? php.Type.array(this.context.phpTypeMapper.convert({ reference: query.valueType }))
+                : this.context.phpTypeMapper.convert({ reference: query.valueType });
+
+            class_.addField(
+                php.field({
+                    name: this.context.getPropertyName(query.name.name),
+                    type,
+                    access: "public",
+                    docs: query.docs
+                })
+            );
+        }
+
+        const service = this.context.getHttpServiceOrThrow(this.serviceId);
+        for (const header of [...service.headers, ...this.endpoint.headers]) {
+            class_.addField(
+                php.field({
+                    name: this.context.getPropertyName(header.name.name),
+                    type: this.context.phpTypeMapper.convert({ reference: header.valueType }),
+                    access: "public",
+                    docs: header.docs
+                })
+            );
+        }
+
+        this.endpoint.requestBody?._visit({
+            reference: (reference) => {
+                class_.addField(
+                    php.field({
+                        name: this.context.getPropertyName(this.wrapper.bodyKey),
+                        type: this.context.phpTypeMapper.convert({ reference: reference.requestBodyType }),
+                        access: "public",
+                        docs: reference.docs
+                    })
+                );
+            },
+            inlinedRequestBody: (request) => {
+                for (const property of [...request.properties, ...(request.extendedProperties ?? [])]) {
```

Comment:
i've been ignoring `extendedProperties` for now

PHP doesn't support multiple inheritance, but we could potentially leverage traits for a similar effect, see [ticket](https://linear.app/buildwithfern/issue/FER-3215/[php]-inheritance)

Reply:
Sounds good - I'll drop them here for now so that we can add them all in at the same time.